rio_tiler/io/stac.py

```diff
@@ -130,7 +139,8 @@ class STACReader(MultiBaseReader):
         include_asset_types (set of string, optional): Only include some assets base on their type.
         exclude_asset_types (set of string, optional): Exclude some assets base on their type.
         reader (rio_tiler.io.BaseReader, optional): rio-tiler Reader. Defaults to `rio_tiler.io.COGReader`.
-        reader_options (dict, optional): additional option to forward to the Reader. Defaults to `{}`.
+        reader_options (dict, optional): Additional option to forward to the Reader. Defaults to `{}`.
+        fetch_options (dict, optional): Options to pass to the client (requests, boto3 ...) fetching the STAC Items. Defaults to `{}`.
```

Comment:
I think somewhere you should document the special behavior for `client` and `request_pays` kwargs for AWS. Something like
"When using a STAC object with HTTPS urls, kwargs is passed to `requests.get` after the url. When using an object with S3 urls, `client` and `request_pays` kwargs are passed to boto3's `get_object`"

Reply:
```suggestion
        fetch_options (dict, optional): Options to pass to `rio_tiler.io.stac.fetch` function fetching the STAC Items. Defaults to `{}`.
```